ParameterHelper.php

```
@@ -0,0 +1,219 @@
+<?php
+
+	namespace Stoic\IO;
+
+	use Stoic\IO\SanitationHelper;
+
+	/**
+	 * Class ParameterHelper
+	 *
+	 * @package Stoic\IO
+	 * @version 1.0.0
+	 */
+	class ParameterHelper {
+
+		/**
+		 * Array of parameters.
+		 *
+		 * @var array
+		 */
+		private $_parameters = array();
+
+		/**
+		 * Sanitation Helper
+		 *
+		 * @var SanitationHelper
+		 */
+		private $_sanitizer = null;
+
+		/**
+		 * Creates a new ParameterHelper instance.
+		 *
+		 * @param array            $params    Array of parameters to dispense.
+		 * @param SanitationHelper $sanitizer Helper class that sanitizes values to a specific type.
+		 */
+		public function __construct(array $params = null, SanitationHelper $sanitizer = null) {
+			if (is_null($params)) {
+				$params = array();
+			}
+
+			if (is_null($sanitizer)) {
+				$sanitizer = new SanitationHelper();
+			}
+
+			$this->_parameters = $params;
+			$this->_sanitizer  = $sanitizer;
+		}
```

I know we had a discussion LONG ago about returns on ctors, but I don't remember where we came down on it?

---

As an additional thought, if we're marking an object as invalid in the ctor, we should consider whether or not it invalidates the object forever.  If it does, we shouldn't return, we should throw an exception so there is an actual failure.

---

Been a long time since commenting on this, and I think here's where I stand today:

There are times when each of these approaches could be argued as 'reasonable', so maybe trying to be consistent is a futile thing.